### PR TITLE
Material Design - M1 & M2 

### PR DESCRIPTION
**Pull request by stojanov-igor**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] This pull request is being made by the same account as the accepted application.
- [x] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [x] In case of acceptance, invoices must be submitted and payments will be transferred to the Polkadot AssetHub and/or fiat account provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/2159

### CODE REVIEW

**Comment by dsm-w3f** [comment]
Hi,  I'll take a look at this evaluation.

**Comment by dsm-w3f** [comment]
@stojanov-igor thank you for the milestone delivery. Please take a look at the two evaluation documents in [this PR](https://github.com/w3f/Grant-Milestone-Delivery/pull/1148) and provide proper answers and fixes. After that, let me know when I can continue this evaluation.

**Comment by stojanov-igor** [comment]
Hi @dsm-w3f 

Thank you for the evaluation and the extensive feedback you have provided. I have resolved the issues mentioned and you can find specifics below:

- I kept the license to be the same as the original license (Unlicense)
- The Dockerfile was updated. There seemed to be an issue with copying _node_modules_ inside the container, but this is now resolved. Documentation was also added to the Readme.
- The two icons are removed from the _AccountSelector_ module. They were there as an example and were not meant to be actionable. The theme toggle is standard in React applications.
- The styling for the _TemplateModule_ has been improved. 
- `yarn test` has been fixed.
- The Cypress tests are meant to serve as an example. I have improved the selectors for the UI elements by adding a new tag to the elements we are testing which shortened the selectors a bit. Thanks for the suggestion.
- I cannot reproduce the misalignment on the copy icons you mentioned. On which browsers/resolution did you notice the discrepancy? 
- There is no additional logic added for the buttons on the _PalletInteractor_. This means that if previously you were able to send the transaction, this should still be the case. Can you check if you are meeting all the conditions for transfer? 

Regarding your other observations, I agree with you. I found a few bugs that exist within the old template. Fixing these was beyond the scope of the work since I focused only on migration to Material Design and Typescript. 

Feel free to continue the evaluation..

**Comment by dsm-w3f** [comment]
@stojanov-igor thank you for the answers and fixes. The M1 is accepted but M2 still have some adjustments to perform. Please see details in the updated evaluation documents on [this PR](https://github.com/w3f/Grant-Milestone-Delivery/pull/1148). Let me know when I can continue this evaluation.

File: deliveries/material_milestone_1_2.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Milestone Delivery :mailbox:
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).**  
+
+* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/si-front-end-template.md 
+* **Milestone Number:**  1 & 2
```
Comment (semuelle) [inline deliveries/material_milestone_1_2.md:6]:
```suggestion
* **Milestone Number:** 1 & 2
```

**Comment by stojanov-igor** [comment]
Hi @dsm-w3f,

Thanks again for the feedback. 

- I have published an [article](https://medium.com/@igorstojanov/substrate-template-with-material-design-typescript-629465de6540) on Medium regarding the grant. 
- I have resolved the runtime error you mentioned. Feel free to verify.

Regarding your comment on Firefox. I use Chrome so the styling is optimized from Chromium-based browsers.

You can proceed with the evaluation. Looking forward to your comments.

**Comment by dsm-w3f** [comment]
@stojanov-igor thank you for the improvements and fixes. We are almost done with this evaluation. There one small bug that still remain. Please see details in the evaluation document in [this PR](https://github.com/w3f/Grant-Milestone-Delivery/pull/1148). Please let me know when I can check it again to finish this evaluation.

**Comment by stojanov-igor** [comment]
Hi @dsm-w3f 

Thanks for the feedback. I have converted the TextField from 'number' to 'text' type. This resolves the last small bug you found. 

Let me know if you notice other issues.

**Comment by dsm-w3f** [comment]
@stojanov-igor thank you for the fix. The two milestones are accepted from my side. [The PR](https://github.com/w3f/Grant-Milestone-Delivery/pull/1148) with the evaluations is open. Great job!

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks, @stojanov-igor and @dsm-w3f. I noticed that the README still refers to the forked repository. Could you update the URLs?

**Comment by stojanov-igor** [comment]
> Thanks, @stojanov-igor and @dsm-w3f. I noticed that the README still refers to the forked repository. Could you update the URLs?

Hi @semuelle. Thank you for your comment. 

Could you please let me know if you're referring to delivery item **0b. Documentation** that [Links](https://github.com/stojanov-igor/substrate-front-end-template/blob/material-design-update/README.md) to the forked repository README? If so, that is the correct updated documentation for the repository. 

Let me know if you meant something else.

**Comment by semuelle** [comment]
> Could you please let me know if you're referring to delivery item **0b. Documentation** that [Links](https://github.com/stojanov-igor/substrate-front-end-template/blob/material-design-update/README.md?rgh-link-date=2024-04-07T20%3A02%3A22Z) to the forked repository README? If so, that is the correct updated documentation for the repository.

This line (and some other lines in the readme) refers to the original repository. Shouldn't it refer to `stojanov-igor/substrate-front-end-template`?

**Comment by stojanov-igor** [comment]
Nice catch @semuelle 

I have updated the README on the branch to reflect the fork and not the original repository. 

I have also created a [pull request](https://github.com/stojanov-igor/substrate-front-end-template/pull/2), to merge the work done to _master branch_ in the forked repository. 

Would you prefer to keep the work in my forked repository, or would also like to create a separate branch on the original repository? Let me know...

**Comment by semuelle** [comment]
> Would you prefer to keep the work in my forked repository, or would also like to create a separate branch on the original repository? Let me know...

I don't know how the maintainers feel about that, but it's worth asking. Perhaps @sacha-l has a minute to review the project and comment.

**Review by semuelle — APPROVED** [review]
I have verified @dsm-w3f's review. Your milestones are hereby accepted, @stojanov-igor. Congrats.

**Pull request merged**